nvflare/app_opt/compression/model_compressor.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union
+
+import numpy as np
+import torch
+from bitsandbytes.functional import quantize_blockwise
+
+from nvflare.apis.dxo import DXO, DataKind, MetaKey
+from nvflare.apis.dxo_filter import DXOFilter
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.app_opt.compression.constant import COMPRESSION_TYPE, DATA_TYPE
+
+
+class ModelCompressor(DXOFilter):
+    def __init__(
+        self,
+        source_data_type="float32",
```

Comment:
in general, why do we need `source_data_type`? Can we not automatically determine it?

Reply:
removed, also removed its counterpart for decompressor